databox/indexer/src/handlers/phraseanet/shared.ts

```diff
@@ -20,45 +20,69 @@ export type AttrDefinitionIndex = Record<
     }
 >;
 
+
+export type TagIndex = Record<
+    number,
+    string
+>;
+
 export type AttrClassIndex = Record<string, AttributeClass>;
 
 export function createAsset(
     workspaceId: string,
     importFiles: boolean,
     record: PhraseanetRecord,
+    rootCollectionPath: string,
+    collectionKeyPrefix: string,
+    key: string,
     collectionName: string,
-    attrDefinitionIndex: AttrDefinitionIndex
+    attrDefinitionIndex: AttrDefinitionIndex,
+    tagIndex: TagIndex,
+    storyCollectionIds: string[]
 ): Asset {
     const document: SubDef | undefined = record.subdefs.find(
         s => s.name === 'document'
     );
 
-    const path = `${escapeSlashes(collectionName)}/${escapeSlashes(
+    const path = `${rootCollectionPath}/${escapeSlashes(collectionName)}/${escapeSlashes(
         record.original_name
     )}`;
 
-    return {
-        workspaceId,
-        key: record.uuid,
-        path,
-        title: record.title,
-        importFile: importFiles,
-        publicUrl: document?.permalink.url,
-        isPrivate: false,
-        attributes: record.caption?.map(c => {
-            const ad = attrDefinitionIndex[c.meta_structure_id.toString()];
-
+    const attributes: AttributeInput[] = [];
+    for(const c of record.caption ?? []) {
+        const ad = attrDefinitionIndex[c.meta_structure_id.toString()];
+        if(ad !== undefined) {
```

Comment:
This should raise an error in my mind

Reply:
used to import a restricted list of fields (faster during dev) - could be removed now but may be useful if we implement "list of fields to import / ignore".